packages/wonder-blocks-form/src/components/text-area.tsx

```diff
@@ -0,0 +1,401 @@
+     */
+    light?: boolean;
+};
+
+const defaultErrorMessage = "This field is required.";
+
+const StyledTextArea = addStyle("textarea");
+
+const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
+    function TextArea(
+        props: TextAreaProps,
+        ref: React.ForwardedRef<HTMLTextAreaElement>,
+    ) {
+        const {
+            onChange,
+            value,
+            placeholder,
+            disabled,
+            id,
+            testId,
+            style,
+            readOnly,
+            autoComplete,
+            name,
+            className,
+            autoFocus,
+            rows,
+            spellCheck,
+            wrap,
+            minLength,
+            maxLength,
+            onClick,
+            onKeyDown,
+            onKeyUp,
+            onFocus,
+            onBlur,
+            validate,
+            onValidate,
+            required,
+            resizeType,
+            light,
+            // Should only include aria related props
+            ...otherProps
+        } = props;
+
+        const [error, setError] = React.useState<string | null>(null);
+
+        const ids = useUniqueIdWithMock("text-area");
+        const uniqueId = id ?? ids.get("id");
+
+        const handleChange = (
+            event: React.ChangeEvent<HTMLTextAreaElement>,
+        ) => {
+            const newValue = event.target.value;
+            onChange(newValue);
+            handleValidation(newValue);
+        };
+
+        const handleValidation = (newValue: string) => {
+            if (validate) {
+                const error = validate(newValue) || null;
+                setError(error);
+                if (onValidate) {
+                    onValidate(error);
+                }
+            } else if (required) {
+                const requiredString =
+                    typeof required === "string"
+                        ? required
+                        : defaultErrorMessage;
+                const error = newValue ? null : requiredString;
+                setError(error);
+                if (onValidate) {
+                    onValidate(error);
+                }
+            }
+        };
+
+        useOnMountEffect(() => {
+            // Only validate on mount if the value is not empty. This is so that fields
+            // don't render an error when they are initially empty
+            if (value !== "") {
+                handleValidation(value);
+            }
+        });
+
+        const getStyles = (): (CSSProperties | Falsy)[] => {
+            // Base styles are the styles that apply regardless of light mode
+            const baseStyles = [
+                styles.textarea,
+                typographyStyles.LabelMedium,
+                resizeType && resizeStyles[resizeType],
+            ];
+            const defaultStyles = [
+                styles.default,
+                !disabled && styles.defaultFocus,
+                disabled && styles.disabled,
```

Comment:
**thought/suggestion:** I wonder if we should rely on the `:disabled` pseudo-class instead. That way we don't need to define that here and instead let `default` define that directly. What do you think?

https://developer.mozilla.org/en-US/docs/Web/CSS/:disabled

Reply:
Great thought! I was thinking this too because it'd be great if we could utilize css specificity! I had tried to do this initially, but got aphrodite typescript errors so I assumed it wasn't supported!

Looking into it again, I see in Wonder Blocks, we explicitly define the types for aphrodite and we don't define the `:disabled` pseudo-class ([WB aphrodite types](https://github.com/Khan/wonder-blocks/blob/68dd6059c24c595c62e07b72b9ec17f0ead2bfbb/types/aphrodite.d.ts#L25-L38)). However, the [types in aphrodite](https://github.com/Khan/aphrodite/blob/225f43c5802259a9e042b384a1f4f2e5b48094ea/typings/css-properties.d.ts#L1822) includes it.

I'm thinking we can create a separate task to update the aphrodite type definitions, or maybe even use the types from aphrodite directly (let me know if there's a reason we define the types in WB!). Then we can also update the styling to use the `:disabled` pseudo-class. Other components could be updated to use it too! What do you think?

Reply:
You could directly add `:disabled` in the WB aphrodite types here, but it's totally fine if you prefer to do it in a separate task. Agreed that a lot of components could be benefited from this :), so probably a separate ticket is the best approach.

Reply:
Will work on this in the follow up ticket to make the TextArea focusable when disabled (WB-1734). We can use a CSS attribute selector then for aria-disabled since we would no longer use the `disabled` attribute!